File: 24/Lua/partTwo.lua
local hailStorm = {}
for line in io.lines("input.txt") do
    local x, y, z, _, _, _ = string.match(line, "^(-?%d+),%s+(-?%d+),%s+(-?%d+)%s+@%s+(-?%d+),%s+(-?%d+),%s+(-?%d+)$")

    local h = { tonumber(x), tonumber(y), tonumber(z) }
    hailStorm[#hailStorm + 1] = h
end

local hailPositionSumCount = {}
for _, hail in ipairs(hailStorm) do
    local sum = 0
    for _, val in ipairs(hail) do
        sum = sum + val
    end
    hailPositionSumCount[sum] = hailPositionSumCount[sum] and hailPositionSumCount[sum] + 1 or 1
end

for sum, count in pairs(hailPositionSumCount) do
    if count > 1 then
        io.write(string.format("%0.f", sum))
    end
end

File: 24/Python/partTwo.py
positionStrings = [line.split()[0:3] for line in open("input.txt").readlines()]
positionList = [
    [int(numberString.replace(",", "")) for numberString in positionString]
    for positionString in positionStrings
]
positionSum = [sum(position) for position in positionList]
result = set([x for x in positionSum if positionSum.count(x) > 1])
print(result.pop())
